### PR TITLE
feat: deduce loader based on file extension

### LINKED ISSUE #3: Any ideas to allow passing configuration?
I specifically mean the `loader` configuration option. In my case, I care about it because the following file doesn't compile with `tsx` (but it does with `ts`):

```ts
// Basically, templated arrow functions are treated as TSX components rather than a templated argument.
const method = <T>() => {
    console.log("hello");
};
```

So maybe... what if we had a `esbuild.config.js(on)` file and read from it, like many other tools?

### PR DESCRIPTION
**Pull request by skyrpex**
Closes #3

### CODE REVIEW

**Comment by antfu** [comment]
@knightly-bot release now

File: src/node.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import sourceMapSupport from 'source-map-support'
 import { transformSync } from 'esbuild'
 import { addHook } from 'pirates'
 import { getOptions } from './options'
+import { FILE } from 'dns'
```
Comment (skyrpex) [inline src/node.ts:7]: @antfu @egoist I don't remember adding this line. Must have autoimported it by error. I think it's of no use.

Comment (antfu) [reply src/node.ts:7]: Could you clean it up so I can request for another build update? Thanks.

**Comment by antfu** [comment]
Could you clean it up so I can request for another build update? Thanks.

**Comment by skyrpex** [comment]
Done!

**Comment by antfu** [comment]
@knightly-bot release now

**Pull request merged**